scipy/stats/_stats_py.py

```diff
@@ -435,18 +452,22 @@ def mode(a, axis=0, nan_policy='propagate'):
         # Fall back to a slower method since np.unique does not work with NaN
         scores = set(np.ravel(a))  # get ALL unique values
         testshape = list(a.shape)
-        testshape[axis] = 1
+        testshape.pop(axis)
```

Comment:
All this code works the same way for `axis=None` as it did before, because `_chk_asarray` ravels and changes `axis` to 0. Previously, the default behavior was to return a 1D array in this case (_not_ an array with the original number of dimensions) and that's what this does now when `keepdims=True`.